vm/x86/x86emu/src/emulator/instruction.rs

```diff
@@ -26,11 +26,11 @@ pub fn address_size(instr: &Instruction) -> usize {
     }
 }
 
-pub fn memory_op_offset(state: &CpuState, instr: &Instruction, operand: u32) -> u64 {
+pub fn memory_op_offset<T: Cpu>(cpu: &mut T, instr: &Instruction, operand: u32) -> u64 {
```

Comment:
This one too. Is there anyway we can keep all these functions non-generic?

Reply:
Addressed in comment chain above, let me know what you think.